pygrocy/grocy_api_client.py

```diff
@@ -327,39 +327,61 @@ def __init__(self, base_url, api_key, port: int = DEFAULT_PORT_NUMBER, verify_ss
                 "GROCY-API-KEY": api_key
             }
 
+    def _do_request(self, request_type: str, end_url: str, data = None):
+        req_url = urljoin(self._base_url, end_url)
+        if request_type == "GET":
+            resp = requests.get(
+                req_url, verify=self._verify_ssl, headers=self._headers)
+        if request_type == "POST":
+            if data:
+                resp = requests.post(
+                    req_url, verify=self._verify_ssl,
+                    headers=self._headers,
+                    data=data)
+        if request_type == "PUT":
+            if data:
+                up_header = self._headers.copy()
+                up_header['accept'] = '*/*'
+                if isinstance(data, dict):
+                    up_header['Content-Type'] = 'application/json'
+                    data = json.dumps(data)
+                else:
+                    up_header['Content-Type'] = 'application/octet-stream'
+                resp = requests.put(
+                    req_url, verify=self._verify_ssl,
+                    headers=up_header,
+                    data=data)
+        resp.raise_for_status()
+        if len(resp.content) > 0:
+            return resp.json()
+
+
     def get_stock(self) -> List[CurrentStockResponse]:
-        req_url = urljoin(self._base_url, "stock")
-        resp = requests.get(req_url, verify=self._verify_ssl, headers=self._headers)
-        if resp.status_code != 200 or not resp.text:
-            return
-        parsed_json = resp.json()
-        return [CurrentStockResponse(response) for response in parsed_json]
+        parsed_json = self._do_request("GET", "stock")
+        if parsed_json:
```

Comment:
The signature claims these methods return a list. If the response is empty it should return an empty list, not None. Imo this if statement should be ommitted for all methods.

Reply:
Thx a lot.
On 6" screen, I can miss some stuff 👍

Reply:
Since an empty is None, I don't understand your statement.
Sorry maybe my lack in Python

Reply:
My bad. Just check.
Thx again 👍

Reply:
Btw, can you tell the tool to check this locally plz ?

Reply:
I don't know what you are trying to do with the `if parsed_json` check.
The current version would still return `None` if `parsed_json` is `None` (heck even if it was `0`!)
I would just write:
```
parsed_json = self._do_request("GET", "stock")
return [CurrentStockResponse(response) for response in parsed_json]
```

If the parsed_json is `None` something went wrong when executing the request, which should be propagated up to the library user (developer) imo, so if the second line fails if `parsed_json` is `None` that would be a correct approach for me.

If, however, you want to catch a `None` value `parsed_json`, then you would have to add an
```
else:
  return []
```
to make sure it always returns a list (empty or filled).

Of course the current approach also works, it just makes it more difficult to use the library as a developer since you have to check for a `None` value separately, instead of being able to rely on receiving a list, or in case of an error get an exception.

Reply:
Thx to explain.
With a ```return []``` the ```if parsed_json:``` doesn't make sense.

Reply:
Ok sorry, it takes time to understand.
I'm correcting.

Reply:
Sorry but your suggestion doesn't work on empty list.
My if statement seems better

Reply:
My bad, something in test.
And seriously, thx to teach 👍

Reply:
I think it's good now